### PR TITLE
Orders types

### PR DESCRIPTION
**Pull request by pannga**

### CODE REVIEW

File: packages/sdk/src/types/Orders.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		metadata?: Record<string, any>
+	}
+	sku?: {
+		override?: boolean
+		sku?: string
+	}
+}
+
+export interface OrdersCreate {
+	customer?: {
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:21]: Please use `CustomersCreateBody` type

File: packages/sdk/src/types/Orders.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		email?: string
+		metadata?: Record<string, any>
+	}
+	amount?: number
+	items?: OrdersItem[]
+	metadata?: Record<string, any>
+}
+
+export interface OrdersCreateResponse {
+	id?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:33]: Please drop `?`

File: packages/sdk/src/types/Orders.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	source_id?: string
+	created_at?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:35]: Please drop `?`

File: packages/sdk/src/types/Orders.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	updated_at?: string
+	status?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:37]: It's one of:
`CREATED | PAID | PROCESSING | CANCELED | FULFILLED`

File: packages/sdk/src/types/Orders.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	amount?: number
+	discount_amount?: number
+	items?: OrdersItem[]
+	metadata?: Record<string, any>
+	customer?: {
+		id?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:43]: If `customer` exists then this exists

File: packages/sdk/src/types/Orders.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+export type OrdersGetResponse = OrdersCreateResponse
+
+export interface OrdersUpdate {
+	id: string
+	source_id?: string
+	status?: 'CREATED' | 'PAID' | 'CANCELLED' | 'FULFILLED'
+	items?: OrdersItem[]
+	amount?: number
+	metadata?: Record<string, any>
+	customer?: {
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:58]: Please use `CustomersCreateBody` type

File: packages/sdk/src/Orders.ts [hunk 7]
```diff
@@ -6,25 +8,28 @@ export class Orders {
...
+			`/orders/${encode(order.id || order.source_id)}`,
+			omit(order, ['id']),
+		)
 	}
 	/**
 	 * @see https://docs.voucherify.io/reference?utm_source=github&utm_medium=sdk&utm_campaign=acq#list-orders
 	 */
-	public list(params: $FixMe) {
-		return this.client.get('/orders', params)
+	public list(params: T.OrdersListParams) {
```
Comment (darekg11) [inline packages/sdk/src/Orders.ts:32]: It should allow to be called without any parameter at all and just default to `{}`

File: packages/sdk/src/types/Orders.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+export type OrdersUpdateResponse = OrdersGetResponse
+
+export interface OrdersListParams {
+	limit?: number
+	page?: number
+}
+
+export interface OrdersListResponse {
+	object?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:71]: Drop `?` and it is always `list`

File: packages/sdk/src/types/Orders.ts [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+	total?: string
```
Comment (darekg11) [inline packages/sdk/src/types/Orders.ts:72]: Drop `?` and change type to `number`

**Review by darekg11 — CHANGES_REQUESTED** [review]
Changes requested:
types definitions

File: packages/sdk/src/types/Orders.ts (hunk 3 above)
Comment (pannga) [reply packages/sdk/src/types/Orders.ts:35]: Dropped only for `created_at`, or should it be dropped for `id` also?

**Review by darekg11 — APPROVED** [review]
Looks good, merging

**Comment by pawelrychlik** [comment]
hurr durr

**Pull request merged**